cql3/expr/expression.cc

```diff
@@ -1028,6 +1028,21 @@ bool has_supporting_index(
                     support);
 }
 
+extern bool index_supports_some_column(
+        const expression& e,
+        const secondary_index::secondary_index_manager& index_manager,
+        allow_local_index allow_local) {
+    single_column_restrictions_map single_col_restrictions = get_single_column_restrictions_map(e);
+
+    for (auto&& [col, col_restrictions] : single_col_restrictions) {
+        if (has_supporting_index(col_restrictions, index_manager, allow_local)) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
```

Comment:
;+1: I think it could be later put in `restrictions.hh` along with other functions that are not meant for generic expressions.
In this PR I didn't want to spend time thinking what should go where, there is enough going on already.